vcpkg/concepts/features.md

```diff
@@ -0,0 +1,157 @@
+---
+title: Understanding features in vcpkg
+description: This article explains the concept of features, their capabilities, and their expected contents in detail.
+author: JavierMatosD
+ms.author: javiermat
+ms.date: 02/21/2024
+ms.topic: concept-article
+---
+
+# Concept: features
+
+## Features
+
+Manifests can define additive sets of functionality, behavior, and dependencies called _features_.
+
+Each feature should be additive with other features: if a project provides features `A` and `B`, then it should build successfully with both `A` and `B` enabled together. Features should not be used to define alternatives. For example, features cannot be used to choose between multiple, exclusive graphics APIs since it is not possible to choose both. Features can depend on other features within same project if the project's manifest has a [`"name"`](../reference/vcpkg-json.md#name).
```

Comment:
What do you think about rewording this paragraph into something like this:

> By design, port features should follow these principles:
> 
> * Additive: enabling feature 'A' should provide new functionality otherwise missing from the package without disabling any other functionality.
> * Non-exclusive: enabling feature 'A' should not prevent feature 'B' from being installed.
>
> As such, features should not be used to define alternative sets of functionalities. For example, a graphics library should not use features to choose between exclusive graphics APIs, since it is not possible to choose both. 
>
> Features can have the following effects on a package's dependencies:
>
> * Add new dependencies, including dependencies on other features from the same package.
> * Enable new features on existing dependencies.